icepyx/core/query.py

```diff
@@ -917,10 +917,12 @@ def visualize_spatial_extent(
         [visual map output]
         """
 
-        world = gpd.read_file(gpd.datasets.get_path("naturalearth_lowres"))
-        f, ax = plt.subplots(1, figsize=(12, 6))
-        world.plot(ax=ax, facecolor="lightgray", edgecolor="gray")
-        geospatial.geodataframe(self.extent_type, self._spat_extent).plot(
-            ax=ax, color="#FF8C00", alpha=0.7
-        )
-        plt.show()
+        from shapely.geometry import Polygon
+        import geoviews as gv
+        gv.extension('bokeh')
+
+        gdf = geospatial.geodataframe(self.extent_type, self._spat_extent)
+        line_geoms = Polygon(gdf['geometry'][0]).boundary
```

Comment:
My understanding is that `self._spat_extent` is already a `shapely.geometry.Polygon` object so it should be possible to get the outline boundary directly like this:

```suggestion
        line_geoms = self._spat_extent.boundary
```

Reply:
Good point! Just checked the `Query` class and found out that `self._spat_extent` is a `shapely.geometry.Polygon` object only when the input `spatial extent` are polygon vertices and polygon file, it will become a list for a bounding box input, so I just kept `Polygon(gdf['geometry'][0]).boundary` for the plot.